core/services/pg/q.go

```diff
@@ -179,5 +218,56 @@ func (q Q) GetNamed(sql string, dest interface{}, arg interface{}) error {
 	}
 	ctx, cancel := q.Context()
 	defer cancel()
-	return errors.Wrap(q.GetContext(ctx, dest, query, args...), "error in get query")
+
+	q.logSqlQuery(query, args...)
+	begin := time.Now()
+	defer q.postSqlLog(ctx, begin)
+
+	return q.withLogError(errors.Wrap(q.GetContext(ctx, dest, query, args...), "error in get query"))
+}
+
+type queryFmt struct {
+	query string
+	args  []interface{}
+}
+
+func (q queryFmt) String() string {
+	if q.args == nil {
+		return q.query
+	}
+	var pairs []string
+	for i, arg := range q.args {
+		pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("%v", arg))
+	}
+	replacer := strings.NewReplacer(pairs...)
+	return replacer.Replace(q.query)
+}
+
+func (q Q) logSqlQuery(query string, args ...interface{}) {
+	if q.config == nil || q.logger == nil {
+		return
+	}
+	if q.config.LogSQL() {
+		q.logger.Debugf("SQL: %s", queryFmt{query, args})
+	}
+}
+
+func (q Q) withLogError(err error) error {
+	if err != nil && err != sql.ErrNoRows && q.logger != nil && q.config.LogSQL() {
```

Comment:
We have a logger nil check here but not above. Do we ever expect it to be nil or is that a programming error?

Reply:
a good catch! I simply forgot to remove these

Reply:
Fixed